Build failure in sysrooted Linux environment (le16toh / be16toh implicit declaration)

Hi tree-sitter maintainers,

I am not a Rust or C expert, but I ran into a build failure involving tree-sitter while trying to package another Rust project (fresh-editor) for conda-forge in this [PR](https://github.com/conda-forge/staged-recipes/pull/31882). I wanted to report this in case it’s a known issue or something that should be handled upstream. It sounds a lot like https://github.com/tree-sitter/tree-sitter/issues/4180 which was solved in https://github.com/tree-sitter/tree-sitter/pull/4181.

## Context

- tree-sitter version: 0.25.10 (pulled via Cargo)
- Platform: Linux
- Build environment: conda-forge / sysrooted toolchain
- Compiler: GCC (C11, strict warnings enabled)

During the build, tree-sitter’s bundled C code fails to compile with errors about implicit function declarations for le16toh and be16toh.

My (possibly incomplete) understanding is:

- tree-sitter’s C sources call le16toh / be16toh
- In minimal or sysrooted Linux environments, these functions are not always declared unless certain headers or feature macros are present
- With modern C standards (-std=c11), implicit declarations are hard errors

This causes the Rust build to fail, even though the Rust code itself seems fine.

I’m not sure what the correct fix should be, and I may be missing something obvious. I’m very open to suggestions or guidance on how this is intended to work in constrained build environments.

Below is the relevant error output for reference:

```bash
 warning: tree-sitter@0.25.10: In file included from $BUILD_PREFIX/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/tree-sitter-0.25.10/src/./lexer.c:3,
 │ │ warning: tree-sitter@0.25.10:                  from $BUILD_PREFIX/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/tree-sitter-0.25.10/src/lib.c:4:
 │ │ warning: tree-sitter@0.25.10: $BUILD_PREFIX/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/tree-sitter-0.25.10/src/././unicode.h: In function 'ts_decode_utf16_le':
 │ │ warning: tree-sitter@0.25.10: $BUILD_PREFIX/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/tree-sitter-0.25.10/src/././unicode.h:18:9: error: implicit declaration of function 'le16
 │ │ toh' [-Wimplicit-function-declaration]
 │ │ warning: tree-sitter@0.25.10:    18 |     (c)=le16toh((s)[(i)++]); \
 │ │ warning: tree-sitter@0.25.10:       |         ^~~~~~~
 │ │ warning: tree-sitter@0.25.10: $BUILD_PREFIX/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/tree-sitter-0.25.10/src/././unicode.h:57:3: note: in expansion of macro 'U16_NEXT_LE'
 │ │ warning: tree-sitter@0.25.10:    57 |   U16_NEXT_LE(((uint16_t *)string), i, length, *code_point);
 │ │ warning: tree-sitter@0.25.10:       |   ^~~~~~~~~~~
 │ │ warning: tree-sitter@0.25.10: $BUILD_PREFIX/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/tree-sitter-0.25.10/src/././unicode.h: In function 'ts_decode_utf16_be':
 │ │ warning: tree-sitter@0.25.10: $BUILD_PREFIX/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/tree-sitter-0.25.10/src/././unicode.h:29:9: error: implicit declaration of function 'be16
 │ │ toh' [-Wimplicit-function-declaration]
 │ │ warning: tree-sitter@0.25.10:    29 |     (c)=be16toh((s)[(i)++]); \
 │ │ warning: tree-sitter@0.25.10:       |         ^~~~~~~
 │ │ warning: tree-sitter@0.25.10: $BUILD_PREFIX/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/tree-sitter-0.25.10/src/././unicode.h:67:3: note: in expansion of macro 'U16_NEXT_BE'
 │ │ warning: tree-sitter@0.25.10:    67 |   U16_NEXT_BE(((uint16_t *)string), i, length, *code_point);
 │ │ warning: tree-sitter@0.25.10:       |   ^~~~~~~~~~~
 │ │ error: failed to run custom build command for `tree-sitter v0.25.10`
 │ │ Caused by:
 │ │   process didn't exit successfully: `$SRC_DIR/target/release/build/tree-sitter-b750bf53ebedd4e1/build-script-build` (exit status: 1)
 │ │   --- stdout
 │ │   cargo:rerun-if-env-changed=CARGO_FEATURE_WASM
 │ │   cargo:rerun-if-changed=$BUILD_PREFIX/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/tree-sitter-0.25.10/src/lib.c
 │ │   cargo:rerun-if-changed=$BUILD_PREFIX/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/tree-sitter-0.25.10/src/subtree.c
 │ │   cargo:rerun-if-changed=$BUILD_PREFIX/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/tree-sitter-0.25.10/src/get_changed_ranges.c
 │ │   cargo:rerun-if-changed=$BUILD_PREFIX/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/tree-sitter-0.25.10/src/reusable_node.h
 │ │   cargo:rerun-if-changed=$BUILD_PREFIX/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/tree-sitter-0.25.10/src/atomic.h
 │ │   cargo:rerun-if-changed=$BUILD_PREFIX/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/tree-sitter-0.25.10/src/wasm
 │ │   cargo:rerun-if-changed=$BUILD_PREFIX/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/tree-sitter-0.25.10/src/clock.h
 │ │   cargo:rerun-if-changed=$BUILD_PREFIX/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/tree-sitter-0.25.10/src/parser.h
 │ │   cargo:rerun-if-changed=$BUILD_PREFIX/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/tree-sitter-0.25.10/src/wasm_store.h
 │ │   cargo:rerun-if-changed=$BUILD_PREFIX/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/tree-sitter-0.25.10/src/alloc.h
 │ │   cargo:rerun-if-changed=$BUILD_PREFIX/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/tree-sitter-0.25.10/src/alloc.c
 │ │   cargo:rerun-if-changed=$BUILD_PREFIX/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/tree-sitter-0.25.10/src/parser.c
 │ │   cargo:rerun-if-changed=$BUILD_PREFIX/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/tree-sitter-0.25.10/src/language.c
 │ │   cargo:rerun-if-changed=$BUILD_PREFIX/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/tree-sitter-0.25.10/src/length.h
 │ │   cargo:rerun-if-changed=$BUILD_PREFIX/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/tree-sitter-0.25.10/src/reduce_action.h
 │ │   cargo:rerun-if-changed=$BUILD_PREFIX/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/tree-sitter-0.25.10/src/node.c
 │ │   cargo:rerun-if-changed=$BUILD_PREFIX/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/tree-sitter-0.25.10/src/lexer.c
 │ │   cargo:rerun-if-changed=$BUILD_PREFIX/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/tree-sitter-0.25.10/src/portable
 │ │   cargo:rerun-if-changed=$BUILD_PREFIX/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/tree-sitter-0.25.10/src/lexer.h
 │ │   cargo:rerun-if-changed=$BUILD_PREFIX/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/tree-sitter-0.25.10/src/point.h
 │ │   cargo:rerun-if-changed=$BUILD_PREFIX/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/tree-sitter-0.25.10/src/stack.h
 │ │   cargo:rerun-if-changed=$BUILD_PREFIX/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/tree-sitter-0.25.10/src/subtree.h
 │ │   cargo:rerun-if-changed=$BUILD_PREFIX/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/tree-sitter-0.25.10/src/tree.c
 │ │   cargo:rerun-if-changed=$BUILD_PREFIX/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/tree-sitter-0.25.10/src/array.h
 │ │   cargo:rerun-if-changed=$BUILD_PREFIX/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/tree-sitter-0.25.10/src/get_changed_ranges.h
 │ │   cargo:rerun-if-changed=$BUILD_PREFIX/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/tree-sitter-0.25.10/src/ts_assert.h
 │ │   cargo:rerun-if-changed=$BUILD_PREFIX/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/tree-sitter-0.25.10/src/host.h
 │ │   cargo:rerun-if-changed=$BUILD_PREFIX/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/tree-sitter-0.25.10/src/tree.h
 │ │   cargo:rerun-if-changed=$BUILD_PREFIX/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/tree-sitter-0.25.10/src/query.c
 │ │   cargo:rerun-if-changed=$BUILD_PREFIX/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/tree-sitter-0.25.10/src/wasm_store.c
 │ │   cargo:rerun-if-changed=$BUILD_PREFIX/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/tree-sitter-0.25.10/src/tree_cursor.h
 │ │   cargo:rerun-if-changed=$BUILD_PREFIX/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/tree-sitter-0.25.10/src/stack.c
 │ │   cargo:rerun-if-changed=$BUILD_PREFIX/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/tree-sitter-0.25.10/src/tree_cursor.c
 │ │   cargo:rerun-if-changed=$BUILD_PREFIX/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/tree-sitter-0.25.10/src/language.h
 │ │   cargo:rerun-if-changed=$BUILD_PREFIX/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/tree-sitter-0.25.10/src/unicode
 │ │   cargo:rerun-if-changed=$BUILD_PREFIX/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/tree-sitter-0.25.10/src/error_costs.h
 │ │   cargo:rerun-if-changed=$BUILD_PREFIX/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/tree-sitter-0.25.10/src/unicode.h
 │ │   OUT_DIR = Some($SRC_DIR/target/x86_64-unknown-linux-gnu/release/build/tree-sitter-489abd0430f3f798/out)
 │ │   OPT_LEVEL = Some(3)
 │ │   TARGET = Some(x86_64-unknown-linux-gnu)
 │ │   CARGO_ENCODED_RUSTFLAGS = Some(-Clink-arg=-Wl,-rpath-link,$PREFIX/lib-Clink-arg=-Wl,-rpath,$PREFIX/lib)
 │ │   HOST = Some(x86_64-unknown-linux-gnu)
 │ │   cargo:rerun-if-env-changed=CC_x86_64-unknown-linux-gnu
 │ │   CC_x86_64-unknown-linux-gnu = None
 │ │   cargo:rerun-if-env-changed=CC_x86_64_unknown_linux_gnu
 │ │   CC_x86_64_unknown_linux_gnu = Some($BUILD_PREFIX/bin/x86_64-conda-linux-gnu-cc)
 │ │   RUSTC_WRAPPER = None
 │ │   cargo:rerun-if-env-changed=CC_ENABLE_DEBUG_OUTPUT
 │ │   cargo:rerun-if-env-changed=CRATE_CC_NO_DEFAULTS
 │ │   CRATE_CC_NO_DEFAULTS = None
 │ │   DEBUG = Some(true)
 │ │   CARGO_CFG_TARGET_FEATURE = Some(fxsr,sse,sse2)
 │ │   cargo:rerun-if-env-changed=CFLAGS
 │ │   CFLAGS = Some(-march=nocona -mtune=haswell -ftree-vectorize -fPIC -fstack-protector-strong -fno-plt -O2 -ffunction-sections -pipe -isystem $PREFIX/include -fdebug-prefix-map=$SRC_DIR=/
 │ │ usr/local/src/conda/fresh-editor-0.1.75 -fdebug-prefix-map=$PREFIX=/usr/local/src/conda-prefix)
 │ │   cargo:rerun-if-env-changed=CC_SHELL_ESCAPED_FLAGS
 │ │   CC_SHELL_ESCAPED_FLAGS = None
 │ │   cargo:rerun-if-env-changed=HOST_CFLAGS
 │ │   HOST_CFLAGS = None
 │ │   cargo:rerun-if-env-changed=CFLAGS_x86_64_unknown_linux_gnu
 │ │   CFLAGS_x86_64_unknown_linux_gnu = Some(-march=nocona -mtune=haswell -ftree-vectorize -fPIC -fstack-protector-strong -fno-plt -O2 -ffunction-sections -pipe -isystem $PREFIX/include -fde
 │ │ bug-prefix-map=$SRC_DIR=/usr/local/src/conda/fresh-editor-0.1.75 -fdebug-prefix-map=$PREFIX=/usr/local/src/conda-prefix)
 │ │   cargo:rerun-if-env-changed=CFLAGS_x86_64-unknown-linux-gnu
 │ │   CFLAGS_x86_64-unknown-linux-gnu = None
 │ │   OUT_DIR = Some($SRC_DIR/target/x86_64-unknown-linux-gnu/release/build/tree-sitter-489abd0430f3f798/out)
 │ │   cargo:rerun-if-env-changed=CC_ENABLE_DEBUG_OUTPUT
 │ │   cargo:rerun-if-env-changed=CRATE_CC_NO_DEFAULTS
 │ │   CRATE_CC_NO_DEFAULTS = None
 │ │   TARGET = Some(x86_64-unknown-linux-gnu)
 │ │   CARGO_CFG_TARGET_FEATURE = Some(fxsr,sse,sse2)
 │ │   HOST = Some(x86_64-unknown-linux-gnu)
 │ │   cargo:rerun-if-env-changed=CFLAGS
 │ │   CFLAGS = Some(-march=nocona -mtune=haswell -ftree-vectorize -fPIC -fstack-protector-strong -fno-plt -O2 -ffunction-sections -pipe -isystem $PREFIX/include -fdebug-prefix-map=$SRC_DIR=/
 │ │ usr/local/src/conda/fresh-editor-0.1.75 -fdebug-prefix-map=$PREFIX=/usr/local/src/conda-prefix)
 │ │   cargo:rerun-if-env-changed=CC_SHELL_ESCAPED_FLAGS
 │ │   CC_SHELL_ESCAPED_FLAGS = None
 │ │   cargo:rerun-if-env-changed=HOST_CFLAGS
 │ │   HOST_CFLAGS = None
 │ │   cargo:rerun-if-env-changed=CFLAGS_x86_64_unknown_linux_gnu
 │ │   CFLAGS_x86_64_unknown_linux_gnu = Some(-march=nocona -mtune=haswell -ftree-vectorize -fPIC -fstack-protector-strong -fno-plt -O2 -ffunction-sections -pipe -isystem $PREFIX/include -fde
 │ │ bug-prefix-map=$SRC_DIR=/usr/local/src/conda/fresh-editor-0.1.75 -fdebug-prefix-map=$PREFIX=/usr/local/src/conda-prefix)
 │ │   cargo:rerun-if-env-changed=CFLAGS_x86_64-unknown-linux-gnu
 │ │   CFLAGS_x86_64-unknown-linux-gnu = None
 │ │   OUT_DIR = Some($SRC_DIR/target/x86_64-unknown-linux-gnu/release/build/tree-sitter-489abd0430f3f798/out)
 │ │   cargo:rerun-if-env-changed=CC_ENABLE_DEBUG_OUTPUT
 │ │   cargo:rerun-if-env-changed=CRATE_CC_NO_DEFAULTS
 │ │   CRATE_CC_NO_DEFAULTS = None
 │ │   TARGET = Some(x86_64-unknown-linux-gnu)
 │ │   CARGO_CFG_TARGET_FEATURE = Some(fxsr,sse,sse2)
 │ │   HOST = Some(x86_64-unknown-linux-gnu)
 │ │   cargo:rerun-if-env-changed=CFLAGS
 │ │   CFLAGS = Some(-march=nocona -mtune=haswell -ftree-vectorize -fPIC -fstack-protector-strong -fno-plt -O2 -ffunction-sections -pipe -isystem $PREFIX/include -fdebug-prefix-map=$SRC_DIR=/
 │ │ usr/local/src/conda/fresh-editor-0.1.75 -fdebug-prefix-map=$PREFIX=/usr/local/src/conda-prefix)
 │ │   cargo:rerun-if-env-changed=CC_SHELL_ESCAPED_FLAGS
 │ │   CC_SHELL_ESCAPED_FLAGS = None
 │ │   cargo:rerun-if-env-changed=HOST_CFLAGS
 │ │   HOST_CFLAGS = None
 │ │   cargo:rerun-if-env-changed=CFLAGS_x86_64_unknown_linux_gnu
 │ │   CFLAGS_x86_64_unknown_linux_gnu = Some(-march=nocona -mtune=haswell -ftree-vectorize -fPIC -fstack-protector-strong -fno-plt -O2 -ffunction-sections -pipe -isystem $PREFIX/include -fde
 │ │ bug-prefix-map=$SRC_DIR=/usr/local/src/conda/fresh-editor-0.1.75 -fdebug-prefix-map=$PREFIX=/usr/local/src/conda-prefix)
 │ │   cargo:rerun-if-env-changed=CFLAGS_x86_64-unknown-linux-gnu
 │ │   CFLAGS_x86_64-unknown-linux-gnu = None
 │ │   OUT_DIR = Some($SRC_DIR/target/x86_64-unknown-linux-gnu/release/build/tree-sitter-489abd0430f3f798/out)
 │ │   cargo:rerun-if-env-changed=CC_ENABLE_DEBUG_OUTPUT
 │ │   cargo:rerun-if-env-changed=CRATE_CC_NO_DEFAULTS
 │ │   CRATE_CC_NO_DEFAULTS = None
 │ │   TARGET = Some(x86_64-unknown-linux-gnu)
 │ │   CARGO_CFG_TARGET_FEATURE = Some(fxsr,sse,sse2)
 │ │   HOST = Some(x86_64-unknown-linux-gnu)
 │ │   cargo:rerun-if-env-changed=CFLAGS
 │ │   CFLAGS = Some(-march=nocona -mtune=haswell -ftree-vectorize -fPIC -fstack-protector-strong -fno-plt -O2 -ffunction-sections -pipe -isystem $PREFIX/include -fdebug-prefix-map=$SRC_DIR=/
 │ │ usr/local/src/conda/fresh-editor-0.1.75 -fdebug-prefix-map=$PREFIX=/usr/local/src/conda-prefix)
 │ │   cargo:rerun-if-env-changed=CC_SHELL_ESCAPED_FLAGS
 │ │   CC_SHELL_ESCAPED_FLAGS = None
 │ │   cargo:rerun-if-env-changed=HOST_CFLAGS
 │ │   HOST_CFLAGS = None
 │ │   cargo:rerun-if-env-changed=CFLAGS_x86_64_unknown_linux_gnu
 │ │   CFLAGS_x86_64_unknown_linux_gnu = Some(-march=nocona -mtune=haswell -ftree-vectorize -fPIC -fstack-protector-strong -fno-plt -O2 -ffunction-sections -pipe -isystem $PREFIX/include -fde
 │ │ bug-prefix-map=$SRC_DIR=/usr/local/src/conda/fresh-editor-0.1.75 -fdebug-prefix-map=$PREFIX=/usr/local/src/conda-prefix)
 │ │   cargo:rerun-if-env-changed=CFLAGS_x86_64-unknown-linux-gnu
 │ │   CFLAGS_x86_64-unknown-linux-gnu = None
 │ │   OUT_DIR = Some($SRC_DIR/target/x86_64-unknown-linux-gnu/release/build/tree-sitter-489abd0430f3f798/out)
 │ │   cargo:rerun-if-env-changed=CC_ENABLE_DEBUG_OUTPUT
 │ │   cargo:rerun-if-env-changed=CRATE_CC_NO_DEFAULTS
 │ │   CRATE_CC_NO_DEFAULTS = None
 │ │   TARGET = Some(x86_64-unknown-linux-gnu)
 │ │   CARGO_CFG_TARGET_FEATURE = Some(fxsr,sse,sse2)
 │ │   HOST = Some(x86_64-unknown-linux-gnu)
 │ │   cargo:rerun-if-env-changed=CFLAGS
 │ │   CFLAGS = Some(-march=nocona -mtune=haswell -ftree-vectorize -fPIC -fstack-protector-strong -fno-plt -O2 -ffunction-sections -pipe -isystem $PREFIX/include -fdebug-prefix-map=$SRC_DIR=/
 │ │ usr/local/src/conda/fresh-editor-0.1.75 -fdebug-prefix-map=$PREFIX=/usr/local/src/conda-prefix)
 │ │   cargo:rerun-if-env-changed=CC_SHELL_ESCAPED_FLAGS
 │ │   CC_SHELL_ESCAPED_FLAGS = None
 │ │   cargo:rerun-if-env-changed=HOST_CFLAGS
 │ │   HOST_CFLAGS = None
 │ │   cargo:rerun-if-env-changed=CFLAGS_x86_64_unknown_linux_gnu
 │ │   CFLAGS_x86_64_unknown_linux_gnu = Some(-march=nocona -mtune=haswell -ftree-vectorize -fPIC -fstack-protector-strong -fno-plt -O2 -ffunction-sections -pipe -isystem $PREFIX/include -fde
 │ │ bug-prefix-map=$SRC_DIR=/usr/local/src/conda/fresh-editor-0.1.75 -fdebug-prefix-map=$PREFIX=/usr/local/src/conda-prefix)
 │ │   cargo:rerun-if-env-changed=CFLAGS_x86_64-unknown-linux-gnu
 │ │   CFLAGS_x86_64-unknown-linux-gnu = None
 │ │   OUT_DIR = Some($SRC_DIR/target/x86_64-unknown-linux-gnu/release/build/tree-sitter-489abd0430f3f798/out)
 │ │   cargo:rerun-if-env-changed=CC_ENABLE_DEBUG_OUTPUT
 │ │   cargo:rerun-if-env-changed=CRATE_CC_NO_DEFAULTS
 │ │   CRATE_CC_NO_DEFAULTS = None
 │ │   TARGET = Some(x86_64-unknown-linux-gnu)
 │ │   CARGO_CFG_TARGET_FEATURE = Some(fxsr,sse,sse2)
 │ │   HOST = Some(x86_64-unknown-linux-gnu)
 │ │   cargo:rerun-if-env-changed=CFLAGS
 │ │   CFLAGS = Some(-march=nocona -mtune=haswell -ftree-vectorize -fPIC -fstack-protector-strong -fno-plt -O2 -ffunction-sections -pipe -isystem $PREFIX/include -fdebug-prefix-map=$SRC_DIR=/
 │ │ usr/local/src/conda/fresh-editor-0.1.75 -fdebug-prefix-map=$PREFIX=/usr/local/src/conda-prefix)
 │ │   cargo:rerun-if-env-changed=CC_SHELL_ESCAPED_FLAGS
 │ │   CC_SHELL_ESCAPED_FLAGS = None
 │ │   cargo:rerun-if-env-changed=HOST_CFLAGS
 │ │   HOST_CFLAGS = None
 │ │   cargo:rerun-if-env-changed=CFLAGS_x86_64_unknown_linux_gnu
 │ │   CFLAGS_x86_64_unknown_linux_gnu = Some(-march=nocona -mtune=haswell -ftree-vectorize -fPIC -fstack-protector-strong -fno-plt -O2 -ffunction-sections -pipe -isystem $PREFIX/include -fde
 │ │ bug-prefix-map=$SRC_DIR=/usr/local/src/conda/fresh-editor-0.1.75 -fdebug-prefix-map=$PREFIX=/usr/local/src/conda-prefix)
 │ │   cargo:rerun-if-env-changed=CFLAGS_x86_64-unknown-linux-gnu
 │ │   CFLAGS_x86_64-unknown-linux-gnu = None
 │ │   cargo:warning=In file included from $BUILD_PREFIX/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/tree-sitter-0.25.10/src/./lexer.c:3,
 │ │   cargo:warning=                 from $BUILD_PREFIX/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/tree-sitter-0.25.10/src/lib.c:4:
 │ │   cargo:warning=$BUILD_PREFIX/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/tree-sitter-0.25.10/src/././unicode.h: In function 'ts_decode_utf16_le':
 │ │   cargo:warning=$BUILD_PREFIX/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/tree-sitter-0.25.10/src/././unicode.h:18:9: error: implicit declaration of function 'le16toh' [-Wimplic
 │ │ it-function-declaration]
 │ │   cargo:warning=   18 |     (c)=le16toh((s)[(i)++]); \
 │ │   cargo:warning=      |         ^~~~~~~
 │ │   cargo:warning=$BUILD_PREFIX/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/tree-sitter-0.25.10/src/././unicode.h:57:3: note: in expansion of macro 'U16_NEXT_LE'
 │ │   cargo:warning=   57 |   U16_NEXT_LE(((uint16_t *)string), i, length, *code_point);
 │ │   cargo:warning=      |   ^~~~~~~~~~~
 │ │   cargo:warning=$BUILD_PREFIX/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/tree-sitter-0.25.10/src/././unicode.h: In function 'ts_decode_utf16_be':
 │ │   cargo:warning=$BUILD_PREFIX/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/tree-sitter-0.25.10/src/././unicode.h:29:9: error: implicit declaration of function 'be16toh' [-Wimplic
 │ │ it-function-declaration]
 │ │   cargo:warning=   29 |     (c)=be16toh((s)[(i)++]); \
 │ │   cargo:warning=      |         ^~~~~~~
 │ │   cargo:warning=$BUILD_PREFIX/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/tree-sitter-0.25.10/src/././unicode.h:67:3: note: in expansion of macro 'U16_NEXT_BE'
 │ │   cargo:warning=   67 |   U16_NEXT_BE(((uint16_t *)string), i, length, *code_point);
 │ │   cargo:warning=      |   ^~~~~~~~~~~
 │ │   --- stderr
 │ │   error occurred in cc-rs: command did not execute successfully (status code exit status: 1): LC_ALL="C" "$BUILD_PREFIX/bin/x86_64-conda-linux-gnu-cc" "-O3" "-ffunction-sections" "-fdata
 │ │ -sections" "-fPIC" "-g" "-gdwarf-4" "-fno-omit-frame-pointer" "-m64" "-I" "$BUILD_PREFIX/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/tree-sitter-0.25.10/src" "-I" "$BUILD_PREFIX
 │ │ /.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/tree-sitter-0.25.10/src/wasm" "-I" "$BUILD_PREFIX/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/tree-sitter-0.25.10/include" 
 │ │ "-w" "-std=c11" "-fvisibility=hidden" "-Wshadow" "-Wno-unused-parameter" "-Wno-incompatible-pointer-types" "-D_POSIX_C_SOURCE=200112L" "-D_DEFAULT_SOURCE" "-D_DARWIN_C_SOURCE" "-march=no
 │ │ cona" "-mtune=haswell" "-ftree-vectorize" "-fPIC" "-fstack-protector-strong" "-fno-plt" "-O2" "-ffunction-sections" "-pipe" "-isystem" "$PREFIX/include" "-fdebug-prefix-map=$SRC_DIR=/usr
 │ │ /local/src/conda/fresh-editor-0.1.75" "-fdebug-prefix-map=$PREFIX=/usr/local/src/conda-prefix" "-march=nocona" "-mtune=haswell" "-ftree-vectorize" "-fPIC" "-fstack-protector-strong" "-fn
 │ │ o-plt" "-O2" "-ffunction-sections" "-pipe" "-isystem" "$PREFIX/include" "-fdebug-prefix-map=$SRC_DIR=/usr/local/src/conda/fresh-editor-0.1.75" "-fdebug-prefix-map=$PREFIX=/usr/local/src/
 │ │ conda-prefix" "-o" "$SRC_DIR/target/x86_64-unknown-linux-gnu/release/build/tree-sitter-489abd0430f3f798/out/98490c8781b409d2-lib.o" "-c" "$BUILD_PREFIX/.cargo/registry/src/index.crates.i
 │ │ o-1949cf8c6b5b557f/tree-sitter-0.25.10/src/lib.c"
```
The full log can be found [here](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=1432901&view=logs&j=4cabe686-70ae-553a-7fd0-310379f2cbac&t=6a4fc7c9-c31a-5115-eff9-6479d72b69ff).

Thanks a lot for your work on tree-sitter, and for taking the time to look at this.
Please let me know if I should provide more details or test something specific.

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -81,7 +81,7 @@ set_target_properties(tree-sitter
                       SOVERSION "${PROJECT_VERSION_MAJOR}.${PROJECT_VERSION_MINOR}"
                       DEFINE_SYMBOL "")
 
-target_compile_definitions(tree-sitter PRIVATE _POSIX_C_SOURCE=200112L _DEFAULT_SOURCE _DARWIN_C_SOURCE)
+target_compile_definitions(tree-sitter PRIVATE _POSIX_C_SOURCE=200112L _DEFAULT_SOURCE _BSD_SOURCE _DARWIN_C_SOURCE)
 
 include(GNUInstallDirs)
 

--- a/Makefile
+++ b/Makefile
@@ -24,7 +24,7 @@ OBJ := $(SRC:.c=.o)
 ARFLAGS := rcs
 CFLAGS ?= -O3 -Wall -Wextra -Wshadow -Wpedantic -Werror=incompatible-pointer-types
 override CFLAGS += -std=c11 -fPIC -fvisibility=hidden
-override CFLAGS += -D_POSIX_C_SOURCE=200112L -D_DEFAULT_SOURCE -D_DARWIN_C_SOURCE
+override CFLAGS += -D_POSIX_C_SOURCE=200112L -D_DEFAULT_SOURCE -D_BSD_SOURCE -D_DARWIN_C_SOURCE
 override CFLAGS += -Ilib/src -Ilib/src/wasm -Ilib/include
 
 # ABI versioning

--- a/Package.swift
+++ b/Package.swift
@@ -27,6 +27,7 @@ let package = Package(
                         .headerSearchPath("src"),
                         .define("_POSIX_C_SOURCE", to: "200112L"),
                         .define("_DEFAULT_SOURCE"),
+                        .define("_BSD_SOURCE"),
                         .define("_DARWIN_C_SOURCE"),
                 ]),
     ],

--- a/build.zig
+++ b/build.zig
@@ -40,6 +40,7 @@ pub fn build(b: *std.Build) !void {
 
     lib.root_module.addCMacro("_POSIX_C_SOURCE", "200112L");
     lib.root_module.addCMacro("_DEFAULT_SOURCE", "");
+    lib.root_module.addCMacro("_BSD_SOURCE", "");
     lib.root_module.addCMacro("_DARWIN_C_SOURCE", "");
 
     if (wasm) {

--- a/crates/xtask/src/build_wasm.rs
+++ b/crates/xtask/src/build_wasm.rs
@@ -199,6 +199,7 @@ pub fn run_wasm(args: &BuildWasm) -> Result<()> {
         "-D", "NDEBUG=",
         "-D", "_POSIX_C_SOURCE=200112L",
         "-D", "_DEFAULT_SOURCE=",
+        "-D", "_BSD_SOURCE=",
         "-D", "_DARWIN_C_SOURCE=",
         "-I", "lib/src",
         "-I", "lib/include",

--- a/lib/binding_rust/build.rs
+++ b/lib/binding_rust/build.rs
@@ -49,6 +49,7 @@ fn main() {
         .include(&include_path)
         .define("_POSIX_C_SOURCE", "200112L")
         .define("_DEFAULT_SOURCE", None)
+        .define("_BSD_SOURCE", None)
         .define("_DARWIN_C_SOURCE", None)
         .warnings(false)
         .file(src_path.join("lib.c"))

